### PR TITLE
RavenDB-7070 detect processor affinity changes in tests

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Server/Commercial/LicenseManager.cs [hunk 1]
```diff
@@ -1064,6 +1067,12 @@ private void SetAffinity(Process process, int cores, bool addPerformanceHint)
             try
             {
                 var currentlyAssignedCores = Bits.NumberOfSetBits(process.ProcessorAffinity.ToInt64());
+                if (IgnoreProcessorAffinityChanges && currentlyAssignedCores != cores)
+                {
+                    Console.WriteLine($"Processor affinity change detected. Current cores: {currentlyAssignedCores}. Requested cores: {cores}.");
```
Comment (ayende) [inline src/Raven.Server/Commercial/LicenseManager.cs:1072]: Why do we want to run `Console.WriteLine` here?

Comment (ppekrol) [reply src/Raven.Server/Commercial/LicenseManager.cs:1072]: It is only used in tests. I can easily check the test output because of that.

Comment (ayende) [reply src/Raven.Server/Commercial/LicenseManager.cs:1072]: It is only used in tests right now, I worry about this changing down the line.

Comment (ppekrol) [reply src/Raven.Server/Commercial/LicenseManager.cs:1072]: After https://issues.hibernatingrhinos.com/issue/RavenDB-14605 is done I will switch this to throw during test execution.

Comment (ayende) [reply src/Raven.Server/Commercial/LicenseManager.cs:1072]: Great!

**Pull request merged**